### PR TITLE
Upgrade Hugo 0.123.6 & fix related warning and error

### LINKED ISSUE #669: Fix Hugo warnings
We have warnings during the build of the site, some of which indicate they may turn into failures in the future due to deprecation.  I counted at least 3 different warnings:
```
make develop
...

WARN  DEPRECATED: Kind "taxonomyterm" used in disableKinds is deprecated, use "taxonomy" instead.
...

WARN  .File.Path on zero object. Wrap it in if or with: {{ with .File }}{{ .Path }}{{ end }}
WARN  .Path when the page is backed by a file is deprecated. We plan to use Path for a canonical source path and you probably want to check the source is a file. To get the current behaviour, you can use a construct similar to the one below:

  {{ $path := "" }}
  {{ with .File }}
	{{ $path = .Path }}
  {{ else }}
	{{ $path = .Path }}
  {{ end }}
...

WARN  found no layout file for "html" for kind "section": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
```

### PR DESCRIPTION
**Pull request by tico88612**
## Which problem is this PR solving?
- Resolves #669 

## Description of the changes
- Upgrade Hugo latest version (0.123.6)
- Fix warning Remove deprecated disableKinds `taxonomyTerm`
- Fix error `.File.Path` nil pointer problem
- Fix warning no layout file for "html" for kind "section"
- Fix warning hugo verbose deprecated
- Fix Hugo 0.123.x remove symlink support
- Fix .Site.IsServer deprecated in Hugo v0.120.0
- Fix data.GetJSON deprecated in Hugo v0.123.0

### FYI

- https://discourse.gohugo.io/t/path-when-the-page-is-backed-by-a-file-is-deprecated/36842
- https://github.com/gohugoio/hugo/issues/11556
- https://github.com/gohugoio/hugo/blob/92684f9a26838a46d1a81e3c250fef5207bcb735/commands/commandeer.go#L138

## How was this change tested?
- No, maybe we can discuss

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,10 +7,10 @@ client-libs-docs:
 	@for d in $(shell ls -d content/docs/*); do \
 		/bin/bash -c "pushd $$d/; \
 			rm -f client-libraries.md client-features.md; \
-			ln -s ../../_client_libs/client-libraries.md; \
-			ln -s ../../_client_libs/client-features.md; \
+			cp ../../_client_libs/client-libraries.md ./; \
+			cp ../../_client_libs/client-features.md ./; \
 			popd"; \
-		echo "sym-linked content/_client_libs/*.md -> $$d/"; \
+		echo "copied content/_client_libs/*.md -> $$d/"; \
```
Comment (yurishkuro) [inline Makefile:13]: why did you need to chance symlinking to copying?

File: themes/jaeger-docs/layouts/partials/docs/panel.html [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+  {{ $currentVersionPath = .Path }}
+{{ end }}
+{{ $currentVersion := index (split $currentVersionPath "/") 1 }}
 {{ $versionDocsPath := printf "content/docs/%s" $currentVersion }}
 {{ $url := .RelPermalink }}
 <div class="nav-wrapper">
   <aside class="nav">
     {{ range site.Pages }}
-      {{ $docVersion := index (split .File.Path "/") 1 }}
+      {{ $docVersionPath := "" }}
```
Comment (yurishkuro) [inline themes/jaeger-docs/layouts/partials/docs/panel.html:14]: there is very similar var versionDocsPath in L9. Here you could use `$pagePath`

File: themes/jaeger-docs/layouts/partials/docs/header.html [hunk 3]
```diff
@@ -1,6 +1,12 @@
 {{ $versions      := site.Params.versions }}
 {{ $latest        := site.Params.latest }}
-{{ $version       := index (split .File.Path "/") 1 }}
+{{ $versionPath := "" }}
```
Comment (yurishkuro) [inline themes/jaeger-docs/layouts/partials/docs/header.html:3]:
```suggestion
{{ $path := "" }}
```

File: themes/jaeger-docs/layouts/partials/docs/panel.html [hunk 4]
```diff
@@ -1,11 +1,23 @@
 <!-- This renders the left-side menu panel -->
-{{ $currentVersion := index (split .File.Path "/") 1 }}
+{{ $currentVersionPath := "" }}
```
Comment (yurishkuro) [inline themes/jaeger-docs/layouts/partials/docs/panel.html:2]:
```suggestion
{{ $path := "" }}
```

File: themes/jaeger-docs/layouts/partials/navbar.html [hunk 5]
```diff
@@ -7,7 +7,13 @@
 {{ $isDocsPage    := eq .Section "docs" }}
 {{ $preview       := getenv "HUGO_PREVIEW" }}
 {{ $isPreview     := eq $preview "true" }}
-{{ $currentVersion := index (split .File.Path "/") 1 }}
+{{ $currentVersionPath := "" }}
```
Comment (yurishkuro) [inline themes/jaeger-docs/layouts/partials/navbar.html:10]:
```suggestion
{{ $path := "" }}
```

File: themes/jaeger-docs/layouts/partials/navbar.html [hunk 6]
```diff
@@ -55,7 +61,13 @@
           </a>
           <div class="navbar-dropdown">
             {{ range $docs }}
-              {{ $docVersion := index (split .File.Path "/") 1 }}
+              {{ $docVersionPath := "" }}
```
Comment (yurishkuro) [inline themes/jaeger-docs/layouts/partials/navbar.html:64]:
```suggestion
              {{ $docPath := "" }}
```

File: themes/jaeger-docs/layouts/partials/search.html [hunk 7]
```diff
@@ -1,4 +1,10 @@
-{{ $version := index (split .File.Path "/") 1 }}
+{{ $versionPath := "" }}
```
Comment (yurishkuro) [inline themes/jaeger-docs/layouts/partials/search.html:1]:
```suggestion
{{ $path := "" }}
```

File: Makefile (hunk 1 above)
Comment (tico88612) [reply Makefile:13]: Because in Hugo 0.123.x symlink won't work for the input files; in other words, it won't generate files.

FYI: https://github.com/gohugoio/hugo/issues/11556

Comment (tico88612) [reply Makefile:13]: If this uses symlinks, this page will be 404.

`https://deploy-preview-675--romantic-neumann-1959d7.netlify.app/docs/<version>/client-features/`

https://deploy-preview-675--romantic-neumann-1959d7.netlify.app/docs/1.54/client-libraries/
https://deploy-preview-675--romantic-neumann-1959d7.netlify.app/docs/1.54/client-features/

Comment (yurishkuro) [reply Makefile:13]: ok. But in this case you should simplify the loop, you don't need rm/pushd/popd, just the copy commands

**Review by yurishkuro — COMMENTED** [review]
when I ran `make develop` on the PR branch I still got the warning abtot `.Path`, is there some page you missed?

**Comment by tico88612** [comment]
I guess it's from here. (It didn't see on my computer.) I would fix it.
https://github.com/jaegertracing/documentation/blob/08b96a308be252c3eac4f592f732e512e59e4dbb/themes/jaeger-docs/layouts/shortcodes/cli/tools-list.html#L1
https://github.com/jaegertracing/documentation/blob/08b96a308be252c3eac4f592f732e512e59e4dbb/themes/jaeger-docs/layouts/shortcodes/currentVersion.html#L1

File: Makefile (hunk 1 above)
Comment (tico88612) [reply Makefile:9]: Done!

**Review by yurishkuro — COMMENTED** [review]
please elaborate why you need a new file themes/jaeger-docs/layouts/_default/section.html

**Comment by tico88612** [comment]
@yurishkuro If there is no such file `themes/jaeger-docs/layouts/_default/section.html`, a warning message will pop up.

```
WARN  found no layout file for "html" for kind "section": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
```

FYI:
- https://discourse.gohugo.io/t/found-no-layout-file-for-html/18983
- https://gohugo.io/templates/lookup-order/

**Comment by yurishkuro** [comment]
I am seeing these additional warnings (which are for some reason tagged as INFO, even though they should break in the future):
```
INFO  deprecated: .Site.IsServer was deprecated in Hugo v0.120.0 and will be removed in a future release. Use hugo.IsServer instead.
INFO  deprecated: data.GetJSON was deprecated in Hugo v0.123.0 and will be removed in a future release. use resources.Get or resources.GetRemote with transform.Unmarshal.
```

**Comment by tico88612** [comment]
> I am seeing these additional warnings (which are for some reason tagged as INFO, even though they should break in the future):
> 
> ```
> INFO  deprecated: .Site.IsServer was deprecated in Hugo v0.120.0 and will be removed in a future release. Use hugo.IsServer instead.
> INFO  deprecated: data.GetJSON was deprecated in Hugo v0.123.0 and will be removed in a future release. use resources.Get or resources.GetRemote with transform.Unmarshal.
> ```

Fixed.
